netket/models/autoreg.py

```diff
@@ -46,19 +47,43 @@ def __post_init__(self):
             raise ValueError("Only unconstrained Hilbert spaces are supported by ARNN.")
 
     @abc.abstractmethod
-    def conditionals(self, inputs: Array, cache: PyTree) -> Tuple[Array, PyTree]:
+    def _conditional(self, inputs: Array, index: int) -> Array:
         """
-        Computes the probabilities for each spin to take each value.
+        Computes the conditional probabilities for a site to take each value.
+        This method gives the expected output only if the correct caches are given in `variables`.
+        Typically, it should only be called successively with indices 0, 1, 2, ...,
+        as in the autoregressive sampling procedure.
 
         Args:
           inputs: configurations with dimensions (batch, Hilbert.size).
-          cache: auxiliary states, e.g., used to implement fast autoregressive sampling.
+          index: index of the site.
 
         Returns:
-          p: the probabilities with dimensions (batch, Hilbert.size, Hilbert.local_size).
-          cache: the updated cache.
+          The probabilities with dimensions (batch, Hilbert.local_size).
+        """
+
+    @abc.abstractmethod
+    def conditionals(self, inputs: Array) -> Array:
+        """
+        Computes the conditional probabilities for each site to take each value.
+
+        Args:
+          inputs: configurations with dimensions (batch, Hilbert.size).
+
+        Returns:
+          The probabilities with dimensions (batch, Hilbert.size, Hilbert.local_size).
+
+        Examples:
+
+          >>> import pytest; pytest.skip("skip automated test of this docstring")
+          >>>
+          >>> p = model.apply(variables, σ, method=model.conditionals)
```

Comment:
can we give instead an example where we call conditionals directly?

Reply:
I think not.
That's how you call a flax function.

Reply:
ah I remember now... this is really some peculiar limitation